src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/DefaultOpenDistroSecurityKeyStore.java

```diff
@@ -223,313 +230,385 @@ private void initSSLConfig() {
+                        : keystorePassword.toCharArray());
+
+                final X509Certificate[] transportKeystoreCert = SSLCertificateHelper.exportServerCertChain(ks,
+                    keystoreAlias);
+                final PrivateKey transportKeystoreKey = SSLCertificateHelper.exportDecryptedKey(ks, keystoreAlias,
+                    (keyPassword == null || keyPassword.length() == 0) ? null
+                        : keyPassword.toCharArray());
+
+                if (transportKeystoreKey == null) {
+                    throw new ElasticsearchException(
+                        "No key found in " + keystoreFilePath + " with alias " + keystoreAlias);
                 }
 
-                final String truststoreType = settings.get(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_TRUSTSTORE_TYPE,
-                        DEFAULT_STORE_TYPE);
-                final String truststorePassword = settings.get(
-                        SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_TRUSTSTORE_PASSWORD,
-                        SSLConfigConstants.DEFAULT_STORE_PASSWORD);
-                final String truststoreAlias = settings
-                        .get(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_TRUSTSTORE_ALIAS, null);
+                if (transportKeystoreCert == null || (transportKeystoreCert != null && transportKeystoreCert.length == 0)) {
+                    throw new ElasticsearchException(
+                        "No certificates found in " + keystoreFilePath + " with alias " + keystoreAlias);
+                }
 
-                try {
+                final KeyStore ts = KeyStore.getInstance(truststoreType);
+                ts.load(new FileInputStream(new File(truststoreFilePath)),
+                    (truststorePassword == null || truststorePassword.length() == 0) ? null
+                        : truststorePassword.toCharArray());
 
-                    final KeyStore ks = KeyStore.getInstance(keystoreType);
-                    ks.load(new FileInputStream(new File(keystoreFilePath)),
-                            (keystorePassword == null || keystorePassword.length() == 0) ? null
-                                    : keystorePassword.toCharArray());
+                final X509Certificate[] trustedTransportCertificates = SSLCertificateHelper
+                    .exportRootCertificates(ts, truststoreAlias);
 
-                    final X509Certificate[] transportKeystoreCert = SSLCertificateHelper.exportServerCertChain(ks,
-                            keystoreAlias);
-                    final PrivateKey transportKeystoreKey = SSLCertificateHelper.exportDecryptedKey(ks, keystoreAlias,
-                            (keyPassword == null || keyPassword.length() == 0) ? null
-                                    : keyPassword.toCharArray());
+                if (trustedTransportCertificates == null || (trustedTransportCertificates != null && trustedTransportCertificates.length == 0)) {
```

Comment:
You don't need to check if it's non-null, since you already checked it in the first part.
Java will only execute the second part of "OR" only if the first condition is true.

```(trustedTransportCertificates == null || trustedTransportCertificates.length == 0```

Reply:
Thanks for the suggestion. Updated this.